### DISCUSSION TITLE
QRCode decode when capturing desktop

### DISCUSSION THREAD

**Discussion by complib2891** [flameshot-org/flameshot#3682] [Ideas]
just a feature request, would be nice since this is a desktop application, when screenshot qrcode, it'll help decode the content ?

**Comment by smallming** [comment]
Related issue https://github.com/flameshot-org/flameshot/issues/511. Will be nice to hear the thoughts of authors and maintainers on this too!

**Comment by mmahmoudian** [comment]
This is duplicate of #511. As indicated in that feature request, the decision was to support this as a plugin. At the moment we have:
1. An RFC (Request for comments) for the plugin system at #2529
2. A basic implementation of a plugin system that supports C++ plugins at #3661
3. A central repository for plugin (not necessarily maintained by devs of Flameshot although they can also write a plugin and add them to the list:  https://github.com/flameshot-org/plugins

At the moment we have a backbone of a plugin system with one sample plugin to add watermark to images. This is still in preliminary state, so feel free to test it, but use it in production at your own risk.

The #511 has some suggestions and discussions that might be useful if you want to write your own plugin. It also contains multiple examples on how to pull off reading qrcode in bash.
